The job **`mingw-check-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15508902877/job/43667226737) [(plain)](https://github.com/rust-lang/rust/commit/79794e56d2f3d9f5dbb4f50f1cc0b5b37489d80d/checks/43667226737/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking test-float-parse v0.1.0 (/checkout/src/tools/test-float-parse)
error[E0277]: the trait bound `f16: FromStr` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:174:13
    |
174 | impl_float!(f16, u16);
    |             ^^^ the trait `FromStr` is not implemented for `f16`
    |
    = help: the following other types implement trait `FromStr`:
              f32
              f64
---
              i8
              isize
            and 6 others

error[E0277]: the trait bound `f16: LowerExp` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:174:13
    |
174 | impl_float!(f16, u16);
    |             ^^^ the trait `LowerExp` is not implemented for `f16`
    |
    = help: the following other types implement trait `LowerExp`:
              f32
              f64
              i128
---
   --> src/tools/test-float-parse/src/traits.rs:105:25
    |
104 | pub trait Float:
    |           ----- required by a bound in this trait
105 |     Copy + fmt::Debug + fmt::LowerExp + FromStr<Err: fmt::Display> + Sized + Send + 'static
    |                         ^^^^^^^^^^^^^ required by this bound in `Float`

error[E0277]: the trait bound `f16: FromStr` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:157:46
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
157 | |                 const MAN_MASK: Self::Int = (Self::Int::ONE << Self::MAN_BITS) - Self::Int::ONE;
    | |                                              ^^^^^^^^^ the trait `FromStr` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `FromStr`:
              f32
              f64
---
              i8
              isize
            and 6 others

error[E0277]: the trait bound `f16: LowerExp` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:157:46
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
157 | |                 const MAN_MASK: Self::Int = (Self::Int::ONE << Self::MAN_BITS) - Self::Int::ONE;
    | |                                              ^^^^ the trait `LowerExp` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `LowerExp`:
              f32
              f64
---
            and 6 others
note: required by a bound in `traits::Float::Int`
   --> src/tools/test-float-parse/src/traits.rs:105:25
    |
105 |     Copy + fmt::Debug + fmt::LowerExp + FromStr<Err: fmt::Display> + Sized + Send + 'static
    |                         ^^^^^^^^^^^^^ required by this bound in `Float::Int`
...
108 |     type Int: Int<Signed = Self::SInt>;
    |          --- required by a bound in this associated type

error[E0277]: the trait bound `f16: FromStr` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:157:64
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
157 | |                 const MAN_MASK: Self::Int = (Self::Int::ONE << Self::MAN_BITS) - Self::Int::ONE;
    | |                                                                ^^^^^^^^^^^^^^ the trait `FromStr` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `FromStr`:
              f32
              f64
---
              i8
              isize
            and 6 others

error[E0277]: the trait bound `f16: LowerExp` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:157:64
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
157 | |                 const MAN_MASK: Self::Int = (Self::Int::ONE << Self::MAN_BITS) - Self::Int::ONE;
    | |                                                                ^^^^ the trait `LowerExp` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `LowerExp`:
              f32
              f64
---
            and 6 others
note: required by a bound in `traits::Float::MAN_BITS`
   --> src/tools/test-float-parse/src/traits.rs:105:25
    |
105 |     Copy + fmt::Debug + fmt::LowerExp + FromStr<Err: fmt::Display> + Sized + Send + 'static
    |                         ^^^^^^^^^^^^^ required by this bound in `Float::MAN_BITS`
...
115 |     const MAN_BITS: u32;
    |           -------- required by a bound in this associated constant

error[E0277]: the trait bound `f16: FromStr` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:157:82
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
157 | |                 const MAN_MASK: Self::Int = (Self::Int::ONE << Self::MAN_BITS) - Self::Int::ONE;
    | |                                                                                  ^^^^^^^^^ the trait `FromStr` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `FromStr`:
              f32
              f64
---
              i8
              isize
            and 6 others

error[E0277]: the trait bound `f16: LowerExp` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:157:82
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
157 | |                 const MAN_MASK: Self::Int = (Self::Int::ONE << Self::MAN_BITS) - Self::Int::ONE;
    | |                                                                                  ^^^^ the trait `LowerExp` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `LowerExp`:
              f32
              f64
---
            and 6 others
note: required by a bound in `traits::Float::Int`
   --> src/tools/test-float-parse/src/traits.rs:105:25
    |
105 |     Copy + fmt::Debug + fmt::LowerExp + FromStr<Err: fmt::Display> + Sized + Send + 'static
    |                         ^^^^^^^^^^^^^ required by this bound in `Float::Int`
...
108 |     type Int: Int<Signed = Self::SInt>;
    |          --- required by a bound in this associated type

error[E0277]: the trait bound `f16: FromStr` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:158:46
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
158 | |                 const SIGN_MASK: Self::Int = Self::Int::ONE << (Self::BITS-1);
    | |                                              ^^^^^^^^^ the trait `FromStr` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `FromStr`:
              f32
              f64
---
              i8
              isize
            and 6 others

error[E0277]: the trait bound `f16: LowerExp` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:158:46
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
158 | |                 const SIGN_MASK: Self::Int = Self::Int::ONE << (Self::BITS-1);
    | |                                              ^^^^ the trait `LowerExp` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `LowerExp`:
              f32
              f64
---
            and 6 others
note: required by a bound in `traits::Float::Int`
   --> src/tools/test-float-parse/src/traits.rs:105:25
    |
105 |     Copy + fmt::Debug + fmt::LowerExp + FromStr<Err: fmt::Display> + Sized + Send + 'static
    |                         ^^^^^^^^^^^^^ required by this bound in `Float::Int`
...
108 |     type Int: Int<Signed = Self::SInt>;
    |          --- required by a bound in this associated type

error[E0277]: the trait bound `f16: FromStr` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:158:65
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
158 | |                 const SIGN_MASK: Self::Int = Self::Int::ONE << (Self::BITS-1);
    | |                                                                 ^^^^^^^^^^ the trait `FromStr` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `FromStr`:
              f32
              f64
---
              i8
              isize
            and 6 others

error[E0277]: the trait bound `f16: LowerExp` is not satisfied
   --> src/tools/test-float-parse/src/traits.rs:158:65
    |
149 | / macro_rules! impl_float {
150 | |     ($($fty:ty, $ity:ty);+) => {
151 | |         $(
152 | |             impl Float for $fty {
...   |
158 | |                 const SIGN_MASK: Self::Int = Self::Int::ONE << (Self::BITS-1);
    | |                                                                 ^^^^ the trait `LowerExp` is not implemented for `f16`
...   |
169 | | }
    | |_- in this expansion of `impl_float!`
...
174 |   impl_float!(f16, u16);
    |   --------------------- in this macro invocation
    |
    = help: the following other types implement trait `LowerExp`:
              f32
              f64
---
            and 6 others
note: required by a bound in `traits::Float::BITS`
   --> src/tools/test-float-parse/src/traits.rs:105:25
    |
105 |     Copy + fmt::Debug + fmt::LowerExp + FromStr<Err: fmt::Display> + Sized + Send + 'static
    |                         ^^^^^^^^^^^^^ required by this bound in `Float::BITS`
...
112 |     const BITS: u32;
    |           ---- required by a bound in this associated constant

For more information about this error, try `rustc --explain E0277`.
```

</details>

--- a/library/core/src/iter/sources/generator.rs
+++ b/library/core/src/iter/sources/generator.rs
@@ -9,8 +9,6 @@
 ///
 /// ```
 /// #![feature(iter_macro, coroutines)]
-/// # #[cfg(not(bootstrap))]
-/// # {
 ///
 /// let it = std::iter::iter!{|| {
 ///     yield 1;
@@ -19,11 +17,10 @@
 /// } }();
 /// let v: Vec<_> = it.collect();
 /// assert_eq!(v, [1, 2, 3]);
-/// # }
 /// ```
 #[unstable(feature = "iter_macro", issue = "none", reason = "generators are unstable")]
 #[allow_internal_unstable(coroutines, iter_from_coroutine)]
-#[cfg_attr(not(bootstrap), rustc_builtin_macro)]
+#[rustc_builtin_macro]
 pub macro iter($($t:tt)*) {
     /* compiler-builtin */
 }

--- a/src/bootstrap/src/core/builder/cargo.rs
+++ b/src/bootstrap/src/core/builder/cargo.rs
@@ -636,6 +636,15 @@ impl Builder<'_> {
         for (restricted_mode, name, values) in EXTRA_CHECK_CFGS {
             if restricted_mode.is_none() || *restricted_mode == Some(mode) {
                 rustflags.arg(&check_cfg_arg(name, *values));
+
+                if *name == "bootstrap" {
+                    // Cargo doesn't pass RUSTFLAGS to proc_macros:
+                    // https://github.com/rust-lang/cargo/issues/4423
+                    // Thus, if we are on stage 0, we explicitly set `--cfg=bootstrap`.
+                    // We also declare that the flag is expected, which we need to do to not
+                    // get warnings about it being unexpected.
+                    hostflags.arg(check_cfg_arg(name, *values));
+                }
             }
         }
 
@@ -645,13 +654,6 @@ impl Builder<'_> {
         if stage == 0 {
             hostflags.arg("--cfg=bootstrap");
         }
-        // Cargo doesn't pass RUSTFLAGS to proc_macros:
-        // https://github.com/rust-lang/cargo/issues/4423
-        // Thus, if we are on stage 0, we explicitly set `--cfg=bootstrap`.
-        // We also declare that the flag is expected, which we need to do to not
-        // get warnings about it being unexpected.
-        hostflags.arg("-Zunstable-options");
-        hostflags.arg("--check-cfg=cfg(bootstrap)");
 
         // FIXME: It might be better to use the same value for both `RUSTFLAGS` and `RUSTDOCFLAGS`,
         // but this breaks CI. At the very least, stage0 `rustdoc` needs `--cfg bootstrap`. See

--- a/src/bootstrap/src/lib.rs
+++ b/src/bootstrap/src/lib.rs
@@ -81,7 +81,10 @@ const LLD_FILE_NAMES: &[&str] = &["ld.lld", "ld64.lld", "lld-link", "wasm-ld"];
 /// (Mode restriction, config name, config values (if any))
 #[expect(clippy::type_complexity)] // It's fine for hard-coded list and type is explained above.
 const EXTRA_CHECK_CFGS: &[(Option<Mode>, &str, Option<&[&'static str]>)] = &[
-    (None, "bootstrap", None),
+    (Some(Mode::Rustc), "bootstrap", None),
+    (Some(Mode::Codegen), "bootstrap", None),
+    (Some(Mode::ToolRustc), "bootstrap", None),
+    (Some(Mode::ToolStd), "bootstrap", None),
     (Some(Mode::Rustc), "llvm_enzyme", None),
     (Some(Mode::Codegen), "llvm_enzyme", None),
     (Some(Mode::ToolRustc), "llvm_enzyme", None),

